boards/bastwan/include/periph_conf.h

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright (C) 2019  Lokotius Filzer
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     boards_bastwan
+ * @{
+ *
+ * @file
+ * @brief       Peripheral MCU configuration for the bastwan board.
+ *
+ * @author      Lokotius Filzer <h-filzer@pixelsalat.de>
+ */
+
+#ifndef PERIPH_CONF_H
+#define PERIPH_CONF_H
+
+#include "periph_cpu.h"
+#include "macros/units.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   GCLK reference speed
+ */
+#define CLOCK_CORECLOCK     MHZ(48)
+
+/**
+ * @brief Enable the internal DC/DC converter
+ *        The board is equipped with the necessary inductor.
+ */
+#define USE_VREG_BUCK       (1)
+
+/**
+ * @name    Timer peripheral configuration
+ * @{
+ */
+static const tc32_conf_t timer_config[] = {
+    {   /* Timer 0 - System Clock */
+        .dev = TC0,
+        .irq = TC0_IRQn,
+        .mclk = &MCLK->APBCMASK.reg,
+        .mclk_mask = MCLK_APBCMASK_TC0 | MCLK_APBCMASK_TC1,
+        .gclk_id = TC0_GCLK_ID,
+        .gclk_src = SAM0_GCLK_8MHZ,
+        .flags = TC_CTRLA_MODE_COUNT32,
+    }
+};
+
+/* Timer 0 configuration */
+#define TIMER_0_CHANNELS    2
+#define TIMER_0_ISR         isr_tc0
+#define TIMER_NUMOF         ARRAY_SIZE(timer_config)
+/** @} */
+
+/**
+ * @name    usart configuration
+ * @{
+ */
+static const uart_conf_t uart_config[] = {
+    {    /* Virtual COM Port */
+        .dev = &SERCOM3->USART,
+        .rx_pin = GPIO_PIN(PA, 18),
+        .tx_pin = GPIO_PIN(PA, 19),
+#ifdef MODULE_PERIPH_UART_HW_FC
+        .rts_pin = GPIO_UNDEF,
+        .cts_pin = GPIO_UNDEF,
+#endif
+        .mux = GPIO_MUX_D,
+        .rx_pad = UART_PAD_RX_2,
+        .tx_pad = UART_PAD_TX_2,
+        .flags = UART_FLAG_NONE,
+        .gclk_src = SAM0_GCLK_MAIN,
+    }
+};
+
+/* interrupt function name mapping */
+#define UART_0_ISR          isr_sercom3
+#define UART_NUMOF          ARRAY_SIZE(uart_config)
+/** @} */
+
+/**
+ * @name    SPI configuration
+ * @{
+ */
+static const spi_conf_t spi_config[] = {
+    {
+        .dev = &(SERCOM4->SPI),
+        .miso_pin = GPIO_PIN(PC, 19),
+        .mosi_pin = GPIO_PIN(PB, 30),
+        .clk_pin = GPIO_PIN(PC, 18),
+        .miso_mux = GPIO_MUX_F,
+        .mosi_mux = GPIO_MUX_F,
+        .clk_mux = GPIO_MUX_F,
```

Comment:
SERCOM MUX on SAML21 are either C or D.

Reply:
spi got totally messed up. reassigned pins, pads and mux in f1315e6b8024d6747af09943413e71b908d4098a

Reply:
GPIO_MUX_F is the internal multiplexed signal for samr34<->semtech see [Datasheet Table 5-2](https://www.mouser.de/datasheet/2/268/SAMR34_R35_Low_Power_LoRa_Sub_GHz_SiP_Data_Sheet_D-1507280.pdf)
reminder to me: do not push after 11pm...

Reply:
I was looking at SAML21 datasheet, I completely forgot about this table. Sorry for the noise.

Reply:
Any hints and suggestions are valuable. I added the 2. spi_config to be able to connect spi periphs to the board so if you wanna x-check, plz..